blocks/library/image/index.js

```diff
@@ -74,46 +75,76 @@ registerBlock( 'core/image', {
 		}
 	},
 
-	edit( { attributes, setAttributes, focus, setFocus } ) {
-		const { url, alt, caption } = attributes;
+	edit: class extends Component {
+		constructor() {
+			super( ...arguments );
+			this.openModal = this.openModal.bind( this );
+			this.onSelect = this.onSelect.bind( this );
+			this.frame = wp.media( {
+				title: wp.i18n.__( 'Select or Upload an image' ),
+				button: {
+					text: wp.i18n.__( 'Select' ),
+				},
+				multiple: false,
+			} );
+
+			// When an image is selected in the media frame...
+			this.frame.on( 'select', this.onSelect );
```

Comment:
Could this be a separate component that block authors can more easily add? It'd be nice if we could keep the complexity of the image block to a minimum, and abstract away the need for hooking up the media modal.